tests/integration_rust/pypi_tests.rs

```diff
@@ -289,3 +289,80 @@ async fn test_allow_insecure_host() {
     .unwrap();
     pixi.update_lock_file().await.unwrap();
 }
+
+#[tokio::test]
+#[cfg_attr(not(feature = "slow_integration_tests"), ignore)]
+async fn test_indexes_are_passed_when_solving_build_pypi_dependencies() {
+    let pypi_indexes = Path::new(env!("CARGO_MANIFEST_DIR")).join("tests/data/pypi-indexes");
+    let pypi_indexes_url = Url::from_directory_path(pypi_indexes.clone()).unwrap();
+
+    let pixi = PixiControl::from_pyproject_manifest(&format!(
+        r#"
+        [project]
+        name = "pypi-build-index"
+        requires-python = ">=3.10"
+        version = "0.1.0"
+
+        [build-system]
+        requires = [
+        "foo",
+        "hatchling",
+        ]
+        build-backend = "hatchling.build"
+
+        [tool.hatch.build]
+        include = ["src"]
+        targets.wheel.strict-naming = false
+        targets.wheel.packages = ["src/pypi_build_index"]
+        targets.sdist.strict-naming = false
+        targets.sdist.packages = ["src/pypi_build_index"]
+
+
+
+        [tool.pixi.project]
+        channels = ["https://prefix.dev/conda-forge"]
+        platforms = ["{platform}"]
+
+        [tool.pixi.dependencies]
+        hatchling = "*"
+
+        [tool.pixi.pypi-options]
+        index-url = "{pypi_indexes}multiple-indexes-a/index"
+        no-build-isolation = ["pypi-build-index"]
```

Comment:
We don't want to store `hatchling` as the source code (first idea discussed with @Hofer-Julian) because it depends on many packages. We can just install it as conda and share the build isolation.